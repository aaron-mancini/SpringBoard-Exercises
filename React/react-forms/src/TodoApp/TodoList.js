import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";
import { v4 as uuidv4 } from "uuid";

const TodoList = () => {
    const [todos, setTodos] = useState([])

    const addTodo = (text) => {
        setTodos([...todos, { text: text, id: uuidv4() }])
    }

    const removeTodo = (id) => {
        setTodos(todos.filter(t => t.id !== id));
    }

    return (
        <div>
            <NewTodoForm addTodo={addTodo} />
            <ul>
                {todos.map(t => <Todo key={t.id} id={t.id} text={t.text} removeTodo={removeTodo}/>)}
            </ul>
        </div>
    )
}

export default TodoList;