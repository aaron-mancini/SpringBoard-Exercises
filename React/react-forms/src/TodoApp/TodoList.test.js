import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";


it("renders without crashing", function() {
    render(<TodoList />);
});

it("matches snapshot", function() {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("creates a new todo", function() {
    const todoList = render(<TodoList />)

    expect(todoList.queryByText("test")).not.toBeInTheDocument();

    const textInput = todoList.getByLabelText("Todo:");
    fireEvent.change(textInput, { target: { value: "test"} })
    const button = todoList.getByText("Add");
    fireEvent.click(button);

    expect(todoList.queryByText("test")).toBeInTheDocument();
})

it("removes a todo", function() {
    const todoList = render(<TodoList />)

    const textInput = todoList.getByLabelText("Todo:");
    fireEvent.change(textInput, { target: { value: "test"} })
    const button = todoList.getByText("Add");
    fireEvent.click(button);

    const removeBtn = todoList.getByText("X");
    fireEvent.click(removeBtn);
    expect(todoList.queryByText("test")).not.toBeInTheDocument();
})