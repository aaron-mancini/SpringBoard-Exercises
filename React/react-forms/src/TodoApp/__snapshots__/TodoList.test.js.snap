// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="text"
      >
        Todo:
      </label>
      <input
        id="text"
        name="text"
        value=""
      />
      <button>
        Add
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
